File: spec/python/test_switch_integers2.py
import unittest

from switch_integers2 import SwitchIntegers2

class TestSwitchIntegers2(unittest.TestCase):
    def test_switch_integers2(self):
        r = SwitchIntegers2.from_file("src/switch_integers.bin")

        self.assertEqual(r.code, 1)
        self.assertEqual(r.len, 7)
        self.assertEqual(r.ham, bytearray([2, 64, 64, 4, 55, 19, 0]))
        self.assertEqual(r.padding, 0)
        self.assertEqual(r.len_mod_str, "13")

File: spec/lua/test_switch_integers2.lua
local luaunit = require("luaunit")

require("switch_integers2")

TestSwitchIntegers2 = {}

function TestSwitchIntegers2:test_switch_integers2()
    local r = SwitchIntegers2:from_file("src/switch_integers.bin")

    luaunit.assertEquals(r.code, 1)
    luaunit.assertEquals(r.len, 7)
    luaunit.assertEquals(r.ham, "\002\064\064\004\055\019\000")
    luaunit.assertEquals(r.padding, 0)
    luaunit.assertEquals(r.len_mod_str, "13")
end
